examples/plotting/plot_rectangle.py

```diff
@@ -0,0 +1,100 @@
+"""
+==========================================================
+Drawing a Rectangle on a map
+==========================================================
+
+In this example we are going to look at how we can plot rectangles on maps
+while describing the dimensions using different methods.
+"""
+
+################################################################################
+# Fundamentally, we will be using the `~sunpy.map.GenericMap.draw_rectangle` method to
+# draw the rectangles. Its function signature is like so:
+#
+# `draw_rectangle(bottom_left, *, width: Unit(‘deg’) = None,
+# height: Unit(‘deg’) = None, axes=None, top_right=None, **kwargs)`
+#
+# All of the extra kwargs are passed on to the `~matplotlib.patches.Rectangle` instance,
+# which allows us to modify the color, linestyle, etc. of the plotted rectangle.
+# A full overview of available options can be found in the Rectangle documentation.
+#
+# Therefore, the different forms in which an input can be provided to the function are:
+#
+# * `~astropy.coordinates.SkyCoord` ((x1, x2), (y1, y2))
+# * `~astropy.coordinates.SkyCoord` bottom_left, `~astropy.coordinates.SkyCoord` top_right
+# * `~astropy.coordinates.SkyCoord` bottom_left, `~astropy.units.Quantity` width, `~astropy.units.Quantity` height
+#
+# Note that from SunPy v2.1, every argument to `~sunpy.map.GenericMap.draw_rectangle` apart from
+# 'bottom_left' (i.e. 'top_right', 'width' and 'height') should be passed as a keyword argument.
```

Comment:
I'm not sure about the specific mention of a version here, especially given that 2.1 hasn't been released here. I think we should just rely on the example code itself and the API docs to communicate this.

Reply:
Depends on if we decide to backport this example or not.

Reply:
Yep that too. On top of that though, it just feels like extraneous info for an example. If I'm a user and just coming to see how to draw a rectangle on a map and I see that the `top_right` kwarg is used, that's all I care about. I'm not concerned about which versions this is/is not supported in.